src/NHibernate/IFutureValueAsync.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading.Tasks;
+
+namespace NHibernate
+{
+	public interface IFutureValueAsync<T>
+	{
+		Task<T> GetValue();
```

Comment:
[Added](https://github.com/OnurGumus/nhibernate-core/commit/181431c84aba942c941106e2f2bfd4975ef8422b) `IFutureEnumerable<T>` and removed `FutureAsync`. It is not perfect, but I also think that is better that having two methods.